### PR TITLE
resolve GHC warnings

### PR DESCRIPTION
**Pull request by glguy**

### CODE REVIEW

File: src/Verifier/SAW/Heapster/SAWTranslation.hs [hunk 1]
```diff
@@ -2727,13 +2733,14 @@ translateSimplImpl (ps0 :: Proxy ps0) mb_simpl m = case mbMatch mb_simpl of
            m
   
   -- Intro for a defined named shape (the other case) is a no-op
-  [nuMP| SImpl_ElimLLVMBlockNamed _ _ nmsh |]
```
Comment (m-yac) [inline src/Verifier/SAW/Heapster/SAWTranslation.hs:2730]: Just curious - why remove this line? and why not also remove line 2709 (`[nuMP| SImpl_IntroLLVMBlockNamed _ _ nmsh |]`) then?

**Review by m-yac — APPROVED** [review]
Looks good to me

File: src/Verifier/SAW/Heapster/SAWTranslation.hs (hunk 1 above)
Comment (glguy) [reply src/Verifier/SAW/Heapster/SAWTranslation.hs:2730]: just an oversight, fixed

File: src/Verifier/SAW/Heapster/SAWTranslation.hs [hunk 2]
```diff
@@ -1632,7 +1636,8 @@ instance TransInfo info =>
     [nuMP| Perm_Fun fun_perm |] ->
       translate fun_perm >>= \tp_term ->
       return $ mkTypeTrans1 tp_term (APTrans_Fun fun_perm)
-
+    [nuMP| Perm_BVProp prop |] ->
+      fmap APTrans_BVProp <$> translate prop
```
Comment (glguy) [inline src/Verifier/SAW/Heapster/SAWTranslation.hs:1640]: Please explicitly confirm that this is the correct implementation

File: src/Verifier/SAW/Heapster/SAWTranslation.hs [hunk 3]
```diff
@@ -632,6 +632,8 @@ instance TransInfo info =>
       return $ error "translate: CharRepr"
     [nuMP| StringRepr UnicodeRepr |] ->
       returnType1 stringTypeOpenTerm
+    [nuMP| StringRepr _ |] ->
+      return $ error "translate: StringRepr non-unicode"
```
Comment (glguy) [inline src/Verifier/SAW/Heapster/SAWTranslation.hs:636]: Do we actually intend/need to handle non-unicode cases?

**Comment by glguy** [comment]
@eddywestbrook we should distinguish the unhandled cases that are expected to be impossible from the ones that are future work. Can you help point out which are which?

File: src/Verifier/SAW/Heapster/SAWTranslation.hs (hunk 2 above)
Comment (glguy) [reply src/Verifier/SAW/Heapster/SAWTranslation.hs:1640]: Eddy said it was what he expected

File: src/Verifier/SAW/Heapster/RustTypes.hs [hunk 4]
```diff
@@ -892,6 +896,7 @@ rsConvertFun w abi (Generics ldefs _tparams@[]
   do arg_shapes <- mapM (rsConvert w) args
      ret_shape <- rsConvert w ret_tp
      layoutFun abi arg_shapes ret_shape
+rsConvertFun _ _ _ _ = fail "rsConvertFun: not supported"
```
Comment (eddywestbrook) [inline src/Verifier/SAW/Heapster/RustTypes.hs]: How about "rsConvertFun: unsupported Rust function type"

File: src/Verifier/SAW/Heapster/SAWTranslation.hs (hunk 3 above)
Comment (eddywestbrook) [reply src/Verifier/SAW/Heapster/SAWTranslation.hs:636]: Not at this point. We only need to support what actually shows up in the Crucible we are pointing Heapster at.

File: src/Verifier/SAW/Heapster/RustTypes.hs (hunk 4 above)
Comment (glguy) [reply src/Verifier/SAW/Heapster/RustTypes.hs]: :+1:

**Review by eddywestbrook — APPROVED** [review]
All of this looks good, modulo one or two tiny comments I added

**Pull request merged**